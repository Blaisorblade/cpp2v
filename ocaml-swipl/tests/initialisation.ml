open Swipl
open Embedding

let _ =
  let built_in () =
    match Versions.get_BUILT_IN () with
    | None    -> Printf.printf "BUILT_IN: not initialised yet\n%!"
    | Some(_) -> Printf.printf "BUILT_IN: initialised\n%!"
  in
  built_in ();
  Printf.printf "Initialised? %b\n%!" (is_initialised ());
  Printf.printf "Initialised? %b\n%!" (is_initialised_argv () = None);
  Printf.printf "Initialising ...\n%!";
  initialise [|Sys.argv.(0); "-q"|];
  Printf.printf "Initialised.\n%!";
  Printf.printf "Initialised? %b\n%!" (is_initialised ());
  let _ =
    match is_initialised_argv () with
    | Some(argv) -> Printf.printf "Initialised with argv = %s\n%!"
                      (String.concat " " (Array.to_list argv))
    | None       -> assert false
  in
  built_in ()
