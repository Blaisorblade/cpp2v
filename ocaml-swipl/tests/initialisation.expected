BUILT_IN: not initialised yet
Initialised? false
Initialised? true
Initialising ...
Initialised.
Initialised? true
Initialised with argv = ./initialisation.exe -q
BUILT_IN: initialised
